### PR TITLE
feat: improve terminal UX with adaptive layout and comprehensive testing

### PR DESCRIPTION
**Pull request by kanywst**
## ✨ Major UX improvements:

- Add minimum terminal size handling (20x6) with clear warnings
- Implement adaptive single/dual pane layout system
- Add intelligent text wrapping and truncation utilities
- Enhance keyboard accessibility (Esc, ?, Tab navigation)
- Improve splash screen with size-based ASCII art variants
- Add contextual status bar with adaptive help text

## 🧪 Comprehensive test coverage:

- 24 new UX test functions covering all terminal sizes
- Test adaptive layout behavior and text handling
- Verify minimum size warnings and error handling
- Ensure layout consistency across operations
- Test keyboard accessibility features

## 🔧 Technical improvements:

- Separate rendering methods for single/dual pane views
- Robust utility functions for text management
- Enhanced error handling for edge cases
- Graceful degradation for small terminals

## ✅ Quality assurance:

- All 33 tests passing
- No breaking changes to existing functionality
- Cross-size compatibility verified (20x6 to unlimited)
- Runtime testing with real certificate data

### CODE REVIEW

File: internal/model/model.go [hunk 1]
```diff
@@ -683,30 +885,56 @@ func (m Model) renderCertificateList(height int) string {
 	for i := start; i < end && i < len(m.certificates); i++ {
 		cert := m.certificates[i]
 
-		// Create compact label for narrow screens
+		// Create adaptive label based on available width
 		var line string
-		if m.width < 40 {
-			// Very compact: just number and status
+		availableWidth := m.width
```
[the remark this answers is not in the document]
Comment (kanywst) [reply internal/model/model.go]: 👍

File: internal/model/splash.go [hunk 2]
```diff
@@ -7,12 +7,39 @@ import (
...
+// renderSplashScreen renders the y509 ASCII art splash screen with adaptive sizing
 func (m Model) renderSplashScreen() string {
 	// Get version dynamically
 	ver := version.GetShortVersion()
 
-	asciiArt := fmt.Sprintf(`
+	var asciiArt string
+	
+	// Adapt ASCII art based on terminal size
+	if m.width < 45 || m.height < 10 {
```
[the remark this answers is not in the document]
Comment (kanywst) [reply internal/model/splash.go]: 👍
